### PR TITLE
an attempt to avoid the problem of mysql going away

### PR DESCRIPTION
**Pull request by peetucket**
Note: this branch should be tested on stage first

## Why was this change made?

another attempt to address #1728

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -50,6 +50,12 @@ Lint/UnusedMethodArgument:
 Metrics/AbcSize:
   Max: 82
 
+# Offense count: 1
```
Comment (peetucket) [inline .rubocop_todo.yml]: This was needed to make rubocop happy for the change in the PR.  Not sure about the other changes to the file.

File: .rubocop.yml [hunk 2]
```diff
@@ -39,3 +39,4 @@ Metrics/BlockLength:
     - 'config/routes.rb'
     - 'spec/**/*'
     - 'lib/tasks/*.rake'
+    - 'app/jobs/create_virtual_objects_job.rb'
```
Comment (peetucket) [inline .rubocop.yml]: This was needed to make rubocop happy for the change in the PR.

File: app/jobs/create_virtual_objects_job.rb [hunk 3]
```diff
@@ -57,6 +57,7 @@ def perform(bulk_action_id, params)
         virtual_objects.reject! { |virtual_object| problematic_druids.include?(virtual_object[:parent_id]) }
       end
 
+      ActiveRecord::Base.clear_active_connections!
       errors = VirtualObjectsCreator.create(virtual_objects: virtual_objects)
       if errors.empty?
```
Comment (justinlittman) [inline app/jobs/create_virtual_objects_job.rb]: You'll want to take a closer look at the code, but I think might want to place it after `VirtualObjectsCreator.create`, as that is the long-running process.

**Comment by justinlittman** [comment]
I'd suggest that we deploy this branch to stage rather than merging.

File: app/jobs/create_virtual_objects_job.rb (hunk 3 above)
Comment (peetucket) [reply app/jobs/create_virtual_objects_job.rb]: I presume this is the long running line: https://github.com/sul-dlss/argo/blob/master/app/services/virtual_objects_creator.rb#L25

I wonder if instead we need to reconnect MySQL here during the polling?  :  https://github.com/sul-dlss/argo/blob/master/app/services/virtual_objects_creator.rb#L43

Comment (peetucket) [reply app/jobs/create_virtual_objects_job.rb]: We need to only deploy argo to stage to test this correct (i.e. no other projects need to be coordinate)?

Comment (justinlittman) [reply app/jobs/create_virtual_objects_job.rb]: Yes, Argo only.

Comment (justinlittman) [reply app/jobs/create_virtual_objects_job.rb]: If my understanding is correct, VirtualObjectsCreator just polls dor-services-app; it doesn't hit the db.

Comment (peetucket) [reply app/jobs/create_virtual_objects_job.rb]: Yeah, but I was thinking that if it sits in that method for a really long time running `redo` while it waits for the response, that is where the connection may get dropped.  But sure, I guess reconnecting once we exit the loop is fine too.

Comment (peetucket) [reply app/jobs/create_virtual_objects_job.rb]: move reconnect below job

Comment (mjgiarlo) [reply app/jobs/create_virtual_objects_job.rb]: @peetucket Can you add a comment above this line so we can remember why we added it? This is great. Thanks!

**Pull request merged**